cleverhans/future/tf2/attacks/deepfool.py

```diff
@@ -0,0 +1,84 @@
+""" The DeepFool attack """
+
+import copy
+import numpy as np
+import tensorflow as tf
+
+
+def deepfool(image_norm, model, num_classes=10, overshoot=0.02, max_iter=50):
```

Comment:
High-level comment: it would be helpful to add some comments to the code to describe what it's doing.